backend/core/src/main/java/org/sonarsource/sonarlint/core/repository/rules/RulesRepository.java

```diff
@@ -100,6 +101,14 @@ public void connectionRemoved(ConnectionConfigurationRemovedEvent e) {
     evictAll(e.getRemovedConnectionId());
   }
 
+  @EventListener
+  public void configScopeRemoved(ConfigurationScopeRemovedEvent e) {
+    var removedBindingConfiguration = e.getRemovedBindingConfiguration();
+    if (removedBindingConfiguration.isBound()) {
+      evictAll(removedBindingConfiguration.getConnectionId());
```

Comment:
Is it possible that there are multiple binging configurations bound to the same connection? E.g. User has two open projects, both of which are connected to the same SQ. 
In this situation, if a user closes one project, we would evict the rules associated with the connection and there would be problems for other project using the same connection.
Do we have to check here that there are no other bindings for that connectionId before evicting?